collector/logs/engine/worker.go

```diff
@@ -52,15 +58,23 @@ func (w *worker) processBatch(ctx context.Context, batch *types.LogBatch) {
 	for _, log := range batch.Logs {
 		log.Freeze()
 	}
-	err = w.Sink.Send(ctx, batch)
-	if err != nil {
-		metrics.LogsCollectorLogsDropped.WithLabelValues(w.SourceName, w.Sink.Name()).Add(float64(len(batch.Logs)))
-		disposeBatch(batch)
-		// TODO skip batch if error is not recoverable
-		// Nack batch?
-		return
+
+	var wg sync.WaitGroup
+	wg.Add(len(w.Sinks))
+	for _, sink := range w.Sinks {
+		go func(sink types.Sink) {
+			defer wg.Done()
+
+			err = sink.Send(ctx, batch)
```

Comment:
```suggestion
			err := sink.Send(ctx, batch)
```
I think this is a race as each sink can modify the err variable, which appears to be declared outside the scope of the go-routines.